src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalStatistics.cs

```diff
@@ -0,0 +1,249 @@
+
+using System.IO;
+
+namespace System.Net.NetworkInformation
+{
+    internal class LinuxIPGlobalStatistics : IPGlobalStatistics
+    {
+        // Information exposed in the snmp (ipv4) and snmp6 (ipv6) files under /proc/net
+        // Each piece corresponds to a data item defined in the MIB-II specification:
+        // http://www.ietf.org/rfc/rfc1213.txt
+        // Each field's comment indicates the name as it appears in the snmp (snmp6) file.
+        // In the snmp6 file, each name is prefixed with "IP6".
+
+        private bool _forwarding; // Forwarding
+        private int _defaultTtl; // DefaultTTL
+
+        private int _inReceives; // InReceives
+        private int _inHeaderErrors; // InHdrErrors
+        private int _inAddressErrors; // InAddrErrors
+        private int _forwardedDatagrams; // ForwDatagrams (IP6OutForwDatagrams)
+        private int _inUnknownProtocols; // InUnknownProtos
+        private int _inDiscards; // InDiscards
+        private int _inDelivers; // InDelivers
+
+        private int _outRequests; // OutRequestscat 
+        private int _outDiscards; // OutDiscards
+        private int _outNoRoutes; // OutNoRoutes
+
+        private int _reassemblyTimeout; // ReasmTimeout
+        private int _reassemblyRequireds; // ReasmReqds
+        private int _reassemblyOKs; // ReasmOKs
+        private int _reassemblyFails; // ReasmFails
+
+        private int _fragmentOKs; // FragOKs
+        private int _fragmentFails; // FragFails
+        private int _fragmentCreates; // FragCreates
+
+        // Miscellaneous IP information, not defined in MIB-II.
+        private int _numRoutes;
+        private int _numInterfaces;
+
+        private LinuxIPGlobalStatistics() { }
+
+        /// <summary>
+        /// Constructs an IPGlobalStatistics object from the snmp4 stats file.
+        /// </summary>
+        public static LinuxIPGlobalStatistics CreateIPv4Statistics()
+        {
+            LinuxIPGlobalStatistics stats = new LinuxIPGlobalStatistics();
+            string fileContents = File.ReadAllText(LinuxNetworkFiles.SnmpV4StatsFile);
+            int firstIpHeader = fileContents.IndexOf("Ip:");
+            int secondIpHeader = fileContents.IndexOf("Ip:", firstIpHeader + 1);
+            int endOfSecondLine = fileContents.IndexOf(Environment.NewLine, secondIpHeader);
+            string ipData = fileContents.Substring(secondIpHeader, endOfSecondLine - secondIpHeader);
+            StringParser parser = new StringParser(ipData, ' ');
+
+            // NOTE: Need to verify that this order is consistent. Otherwise, we need to parse the first-line header
+            // to determine the order of information contained in the file.
+
+            parser.MoveNextOrFail(); // Skip Ip:
+
+            // According to RFC 1213, "1" indicates "acting as a gateway". "2" indicates "NOT acting as a gateway".
+            stats._forwarding = parser.MoveAndExtractNext() == "1";
```

Comment:
Seems like this code isn't properly encapsulated: (I hate seeing that `._` :) ). 
Please consider moving this within the class and leaving this function with a single responsibility: to create the object (the Factory Method pattern).